app/src/main/java/org/sopt/and/feature/my/MyViewModel.kt

```diff
@@ -11,24 +11,34 @@ import kotlinx.coroutines.flow.asSharedFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.launch
 import org.sopt.and.R
-import org.sopt.and.domain.repository.WavveRepository
+import org.sopt.and.domain.usecase.MyUseCase
+import org.sopt.and.feature.my.MyContract.MySideEffect
+import org.sopt.and.feature.my.MyContract.MyState
 import javax.inject.Inject
 
 @HiltViewModel
 class MyViewModel @Inject constructor(
-    private val wavveRepository: WavveRepository
+    private val getHobbyUseCase: MyUseCase
 ) : ViewModel() {
-    private val _state = MutableStateFlow(MyState())
+    private val _state: MutableStateFlow<MyState>
+        get() = MutableStateFlow(MyState())
+    private val currentState: MyState
+        get() = state.value
+
     val state: StateFlow<MyState>
         get() = _state.asStateFlow()
 
     private val _sideEffect: MutableSharedFlow<MySideEffect> = MutableSharedFlow()
     val sideEffect: SharedFlow<MySideEffect>
         get() = _sideEffect.asSharedFlow()
 
+    private fun setState(reduce: MyState.() -> MyState) {
+        _state.value = currentState.reduce()
+    }
+
     fun getUserHobby() {
         viewModelScope.launch {
-            wavveRepository.getHobby().onSuccess { hobbyEntity ->
+            getHobbyUseCase.invoke().onSuccess { hobbyEntity ->
```

Comment:
```suggestion
            getHobbyUseCase().onSuccess { hobbyEntity ->
```
이렇게도 사용할 수 있을 것 같아요.
invoke 함수가 무엇인지 공부해 봅시다.